src/FactorAPI/ComputeEngineIntegrationTest.java

```diff
@@ -0,0 +1,29 @@
+package factorapi;
+
+import org.junit.jupiter.api.Test;
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class ComputeEngineIntegrationTest {
+    @Test
+    void testComputeEngineIntegration() {
+        List<Integer> inputList = List.of(1, 10, 25);
+        List<String> outputList = new ArrayList<>();
+
+        InMemoryInputConfig inputConfig = new InMemoryInputConfig(inputList);
+        InMemoryOutputConfig outputConfig = new InMemoryOutputConfig(outputList);
+        InMemoryDataStore dataStore = new InMemoryDataStore(inputConfig, outputConfig);
+        
+        ComputeJobRequest request = new ComputeJobRequestImpl("source", ",", "destination", "type");
+        ComputeEngine computeEngine = new ComputeEngineImpl();
+
+        computeEngine.initializeJob(request);
+        computeEngine.executeJob(request);
+
+        dataStore.processData(); // Simulate processing
+
```

Comment:
Here, I think the dataStore would need to be passed the result from the compute engine. If the result is a 
List<String> result, what could work is dataStore.write(result). I'm gonna rework mine so that this will write the result to the outputConfig object.